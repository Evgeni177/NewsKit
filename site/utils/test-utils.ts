import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme = renderToFragmentWithThemeFactory(
  docsThemeLight,
);
