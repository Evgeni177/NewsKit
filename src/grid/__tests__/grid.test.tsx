import {renderToFragmentWithTheme} from '../../test/test-utils';
import {Grid} from '..';

describe('Grid Container', () => {
  test('renders expected styles', () => {
    const fragment = renderToFragmentWithTheme(Grid);
    expect(fragment).toMatchSnapshot();
  });

  test('renders expected column styles', () => {
    const fragment = renderToFragmentWithTheme(Grid, {
      xsColumnGutter: 'space000',
      smColumnGutter: 'space010',
      mdColumnGutter: 'space030',
      lgColumnGutter: 'space050',
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders expected row styles', () => {
    const fragment = renderToFragmentWithTheme(Grid, {
      xsRowGutter: 'space000',
      smRowGutter: 'space010',
      mdRowGutter: 'space030',
      lgRowGutter: 'space050',
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders expected max width', () => {
    const fragment = renderToFragmentWithTheme(Grid, {
      maxWidth: '500px',
    });
    expect(fragment).toMatchSnapshot();
  });
});
