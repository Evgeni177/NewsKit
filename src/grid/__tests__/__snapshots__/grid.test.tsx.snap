// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Container renders expected column styles 1`] = `
<DocumentFragment>
  .emotion-1 {
  width: 100%;
  max-width: 1920px;
  margin: 0 auto;
}

.emotion-0 {
  box-sizing: border-box;
  background-clip: padding-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

@media screen {
  .emotion-0 {
    margin: -16px 16px 0 16px;
  }
}

@media screen and (min-width:480px) {
  .emotion-0 {
    margin: -16px 14px 0 14px;
  }
}

@media screen and (min-width:768px) {
  .emotion-0 {
    margin: -0px -6px 0 -6px;
  }
}

@media screen and (min-width:1024px) {
  .emotion-0 {
    margin: -24px 12px 0 12px;
  }
}

<div
    class="emotion-1"
  >
    <div
      class="emotion-0"
    />
  </div>
</DocumentFragment>
`;

exports[`Grid Container renders expected max width 1`] = `
<DocumentFragment>
  .emotion-1 {
  width: 100%;
  max-width: 1920px;
  margin: 0 auto;
}

.emotion-0 {
  box-sizing: border-box;
  background-clip: padding-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

@media screen {
  .emotion-0 {
    margin: -16px 8px 0 8px;
  }
}

@media screen and (min-width:480px) {
  .emotion-0 {
    margin: -16px 8px 0 8px;
  }
}

@media screen and (min-width:768px) {
  .emotion-0 {
    margin: -0px 0px 0 0px;
  }
}

@media screen and (min-width:1024px) {
  .emotion-0 {
    margin: -24px 12px 0 12px;
  }
}

<div
    class="emotion-1"
  >
    <div
      class="emotion-0"
    />
  </div>
</DocumentFragment>
`;

exports[`Grid Container renders expected row styles 1`] = `
<DocumentFragment>
  .emotion-1 {
  width: 100%;
  max-width: 1920px;
  margin: 0 auto;
}

.emotion-0 {
  box-sizing: border-box;
  background-clip: padding-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

@media screen {
  .emotion-0 {
    margin: -0px 8px 0 8px;
  }
}

@media screen and (min-width:480px) {
  .emotion-0 {
    margin: -4px 8px 0 8px;
  }
}

@media screen and (min-width:768px) {
  .emotion-0 {
    margin: -12px 0px 0 0px;
  }
}

@media screen and (min-width:1024px) {
  .emotion-0 {
    margin: -24px 12px 0 12px;
  }
}

<div
    class="emotion-1"
  >
    <div
      class="emotion-0"
    />
  </div>
</DocumentFragment>
`;

exports[`Grid Container renders expected styles 1`] = `
<DocumentFragment>
  .emotion-1 {
  width: 100%;
  max-width: 1920px;
  margin: 0 auto;
}

.emotion-0 {
  box-sizing: border-box;
  background-clip: padding-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

@media screen {
  .emotion-0 {
    margin: -16px 8px 0 8px;
  }
}

@media screen and (min-width:480px) {
  .emotion-0 {
    margin: -16px 8px 0 8px;
  }
}

@media screen and (min-width:768px) {
  .emotion-0 {
    margin: -0px 0px 0 0px;
  }
}

@media screen and (min-width:1024px) {
  .emotion-0 {
    margin: -24px 12px 0 12px;
  }
}

<div
    class="emotion-1"
  >
    <div
      class="emotion-0"
    />
  </div>
</DocumentFragment>
`;
